File: 3way/problem8/solution.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var increasingTriplet = function(nums) {
    if(nums.length < 3) return false;
    let min1 = Infinity;
    let min2 = Infinity;
    for(let n of nums){
        if(n <= min1) min1 = n;
        else if(n <= min2) min2 = n;
        else return true;
    }

    return false;
};

File: 3way/problem8/solution.c
#include <limits.h>

bool increasingTriplet(int* nums, int numsSize) {
    if(numsSize < 3 || nums == NULL){
        return false;
    }
    int min1 = INT_MAX;
    int min2 = INT_MAX;
    for(int i = 0; i < numsSize; i++){
        if(nums[i] <= min1){
            min1 = nums[i];
        }else if(nums[i] <= min2){
            min2 = nums[i];
        }else{
            return true;
        }
    }

    return false;
}
